.github/workflows/build-oci.yaml

```diff
@@ -9,56 +25,70 @@ on:
     branches: [ main ]
 
 jobs:
-  build:
-    name: build
-    runs-on: ubuntu-24.04
+  build-and-upload:
+    name: Build and Upload Images
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-24.04, ubuntu-24.04-arm]
     steps:
-      - name: Prepare runner for build multi arch
-        shell: bash
-        run: |
-          sudo apt-get update && sudo apt-get install -y qemu-user-static
-
       - name: Checkout code
-        uses: actions/checkout@v4
-
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Install Podman (only for arm64)
+        if: contains(matrix.os, 'arm')
+        run: |
+          sudo apt-get update -y
+          sudo apt-get -y install podman
+      - name: Set environment variables based on architecture
+        run: |
+          if [[ "${{ matrix.os }}" == *"arm"* ]]; then
+            echo "ARCH_TYPE=arm64" >> $GITHUB_ENV
+          else
+            echo "ARCH_TYPE=amd64" >> $GITHUB_ENV
+          fi
       - name: Build image for PR
-        if: github.event_name == 'pull_request' 
+        if: github.event_name == 'pull_request'
         env:
           IMG: ghcr.io/redhat-developer/mapt:pr-${{ github.event.number }}
         shell: bash
         run: |
-          make oci-build
-          make oci-save
+          make oci-build-${{ env.ARCH_TYPE }}
+          make oci-save-${{ env.ARCH_TYPE }}
           echo ${IMG} > mapt-image
 
       - name: Build and Push image for Release
         if: github.event_name == 'push'
         run: |
-          make oci-build
-          make oci-save
+          make oci-build-${{ env.ARCH_TYPE }}
+          make oci-save-${{ env.ARCH_TYPE }}
 
       - name: Upload mapt artifacts for PR
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
         with:
-          name: mapt
-          path: mapt*
+          name: mapt-${{ env.ARCH_TYPE }}
+          path: |
+            mapt-image
+            mapt-${{ env.ARCH_TYPE }}*
 
   push:
     name: push
-    if: github.event_name == 'push' 
```

Comment:
I think the `if: github.event_name == 'push' ` in the push action is needed otherwise it will try to push on PR and it fail due to credentials missing...actually this is the current failure you can see on the PR